File: library/src/main/java/com/afollestad/iconrequest/RemoteValidator.java
package com.afollestad.iconrequest;

import android.support.annotation.NonNull;
import com.afollestad.bridge.Response;
import com.afollestad.bridge.ResponseValidator;
import org.json.JSONObject;

/** @author Aidan Follestad (afollestad) */
class RemoteValidator extends ResponseValidator {

  @SuppressWarnings("ConstantConditions")
  @Override
  public boolean validate(@NonNull Response response) throws Exception {
    String body = response.asString();
    if (body != null && body.startsWith("{")) {
      JSONObject json = response.asJsonObject();
      if (!json.getString("status").equals("success")) throw new Exception(json.getString("error"));
    }
    return true;
  }

  @NonNull
  @Override
  public String id() {
    return "backend-validator";
  }
}

File: library/src/main/java/com/afollestad/iconrequest/RemoteValidator.kt
package com.afollestad.iconrequest

import com.afollestad.bridge.Response
import com.afollestad.bridge.ResponseValidator
import org.json.JSONObject

/** @author Aidan Follestad (afollestad) */
internal class RemoteValidator : ResponseValidator() {
  @Throws(Exception::class)
  override fun validate(response: Response): Boolean {
    val body = response.asString()
    if (body != null && body.startsWith("{")) {
      val json = response.asJsonObject()
      if (json!!.getString("status") != "success") throw Exception(json.getString("error"))
    }
    return true
  }

  override fun id(): String {
    return "backend-validator"
  }
}
